cardano-crypto-class/src/Cardano/Crypto/Hash/Blake2b.hs

```diff
@@ -10,19 +10,41 @@ module Cardano.Crypto.Hash.Blake2b
   )
 where
 
+import Control.Monad (unless)
+import Cardano.Crypto.Libsodium.C
+
 import Cardano.Crypto.Hash.Class
-import qualified "cryptonite" Crypto.Hash as H
-import qualified Data.ByteArray as BA
+import Foreign.ForeignPtr (withForeignPtr, mallocForeignPtrBytes)
+import Foreign.Ptr (castPtr, nullPtr)
+import Foreign.C.Error (errnoToIOError, getErrno)
+import Data.Proxy
+import GHC.TypeLits (natVal)
+import GHC.IO.Exception (ioException)
+import System.IO.Unsafe (unsafePerformIO)
+
+import qualified Data.ByteString as B
 
 data Blake2b_224
 data Blake2b_256
 
 instance HashAlgorithm Blake2b_224 where
   type SizeHash Blake2b_224 = 28
   hashAlgorithmName _ = "blake2b_224"
-  digest _ = BA.convert . H.hash @_ @H.Blake2b_224
+  digest _ = blake2b_libsodium (fromIntegral (natVal (Proxy::Proxy (SizeHash Blake2b_256))))
```

Comment:
Is this right? It says `SizeHash Blake2b_256` but surely that should be 224 not 256?

If this comes down to an integer, wouldn't it be simpler to use just `224` rather than `(fromIntegral (natVal (Proxy::Proxy (SizeHash Blake2b_224))))`?